File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/movie/model/MovieReleaseDates.java
package dev.reuss.tmdb.domain.movie.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import dev.reuss.tmdb.common.TmdbModel;
import dev.reuss.tmdb.util.TmdbCollections;

import java.util.List;

/**
 * Release dates for a TMDB movie.
 *
 * @param id      TMDB movie id
 * @param results release dates grouped by country
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record MovieReleaseDates(
        int id,
        List<MovieReleaseDateRegion> results
) implements TmdbModel {

    public MovieReleaseDates {
        results = TmdbCollections.list(results);
    }
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/movie/model/MovieReleaseDates.kt
package dev.reuss.tmdb.domain.movie.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel

/**
 * Release dates for a TMDB movie.
 *
 * @property id TMDB movie id
 * @property results release dates grouped by country
 */
@JvmRecord
data class MovieReleaseDates(
    @all:JsonProperty("id")
    val id: Int,
    @all:JsonProperty("results")
    val results: List<MovieReleaseDateRegion> = emptyList(),
) : TmdbModel
